model/map/src/main/java/org/keycloak/models/map/user/MapUserProvider.java

```diff
@@ -559,6 +559,91 @@ public int getUsersCount(RealmModel realm, boolean includeServiceAccount) {
         return (int) storeWithRealm(realm).getCount(withCriteria(mcb));
     }
 
+    @Override
+    public int getUsersCount(RealmModel realm, Map<String, String> attributes) {
+        LOG.tracef("getUsersCount(%s, %s)%s", realm, attributes, getShortStackTrace());
+
+        final DefaultModelCriteria<UserModel> mcb = criteria();
+        DefaultModelCriteria<UserModel> criteria = mcb.compare(SearchableFields.REALM_ID, Operator.EQ, realm.getId());
+
+        final boolean exactSearch = Boolean.parseBoolean(attributes.getOrDefault(UserModel.EXACT, Boolean.FALSE.toString()));
+
+        for (Map.Entry<String, String> entry : attributes.entrySet()) {
```

Comment:
Currently methods `getUsersCount` and `searchForUserStream` contains the same code, could you please refactor it to a separate method?